core/src/main/java/cz/o2/proxima/scheme/JavaSerializer.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2017-2019 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.scheme;
+
+import cz.o2.proxima.util.Classpath;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Value serializer for java classes.
+ *
+ * Class must implements ${@link Serializable} interface.
+ * Examples:
+ *  scheme: "java:cz.o2.proxima.model.Event"
+ *  scheme: "java:String"
+ */
+@Slf4j
+public class JavaSerializer implements ValueSerializerFactory {
+  private final Map<URI, ValueSerializer<?>> cache = new ConcurrentHashMap<>();
+  private static final List<String> javaPackages = Arrays.asList("java.lang");
+
+  @Override
+  public String getAcceptableScheme() {
+    return "java";
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public <T> ValueSerializer<T> getValueSerializer(URI specifier) {
+    return (ValueSerializer<T>)
+        cache.computeIfAbsent(specifier, JavaSerializer::createSerializer);
+  }
+
+  private static <T> ValueSerializer<T> createSerializer(URI scheme) {
+    return new ValueSerializer<T>() {
```

Comment:
Sorry, one (hopefully last) comment. :)
I'd log here something like
```
log.warn("Using JavaSerializer for URI {}. This can result in low performance, please use with caution.", scheme);
```

Reply:
Added.